### ISSUE TITLE
Update optional output directory from application

### ISSUE THREAD

**Issue by nreinicke** [NREL/routee-compass#32]
In #22 we added a new output plugin that writes the results of the application to an output file and that output file is specified in the config for the output plugin. This is inconvenient if you want to run batches of queries and save them to different files.

One idea to address this would be to expose a method on the compass application that attempts to set the output directory for the to disk plugin. It would have to scan the output plugins that have been loaded into the app and then either err if the to disk output plugin isn't loaded or modify it's output directory. Then, users could load the application like:

```python
app = CompassApp.from_config_file("my/config.toml").with_output_directory("path/to/outputs.json")

app = app.with_output_directory("new/path/to/outputs.json")
```

**Comment by nreinicke** [comment]
I took a brief meander down this path and pulled back when I realized that we can't just scan the output plugins and find if the to_disk plugin is loaded since we hold the plugins as a `Box<dyn OutputPlugin>` type. Some other ideas:

 - pass an output_file argument to the `CompassApp.from_config_file` function that will overwrite any output file parameter set in the config
 - pass an output_file argument as part of the query that gets read by the to_disk plugin and either appends a result to an existing file or opens a new file if one doesn't exist. 

Neither of these feel great and maybe we should continue thinking about other options.
